modules/apps/adaptive-media/adaptive-media-content-transformer-impl/src/main/java/com/liferay/adaptive/media/content/transformer/internal/HtmlContentTransformerImpl.java

```diff
@@ -39,42 +37,90 @@ public String transform(String html) throws PortalException {
 			return null;
 		}
 
-		if (!html.contains(AMImageHTMLConstants.ATTRIBUTE_NAME_FILE_ENTRY_ID)) {
-			return html;
-		}
+		StringBundler sb = new StringBundler();
 
-		return super.transform(html);
-	}
+		int i = 0;
 
-	@Override
-	protected FileEntry getFileEntry(Matcher matcher) throws PortalException {
-		long fileEntryId = GetterUtil.getLong(matcher.group(1));
+		while (i < html.length()) {
+			int imgStart = html.indexOf(_OPEN_TAG_TOKEN_IMG, i);
 
-		return _dlAppLocalService.getFileEntry(fileEntryId);
-	}
+			if (imgStart == -1) {
+				sb.append(html.substring(i));
 
-	@Override
-	protected Pattern getPattern() {
-		return _pattern;
-	}
+				break;
+			}
 
-	@Override
-	protected String getReplacement(String originalImgTag, FileEntry fileEntry)
-		throws PortalException {
+			int imgEnd = html.indexOf(
+				CharPool.GREATER_THAN, imgStart + _OPEN_TAG_TOKEN_IMG.length());
 
-		return _amImageHTMLTagFactory.create(originalImgTag, fileEntry);
-	}
+			if (imgEnd == -1) {
+				sb.append(html.substring(i));
 
-	@Override
-	protected boolean isSupported(FileEntry fileEntry) {
-		return _amImageMimeTypeProvider.isMimeTypeSupported(
-			fileEntry.getMimeType());
+				break;
+			}
+
+			imgEnd++;
+
+			int attributeStart = html.indexOf(
```

Comment:
Suggestion

I'd rename  attributeStart var to "fileEntryIdAttributeStart", and fileEntryIdStart to "fileEntryIdAttributeValueStart" in order to make it more readable